ansible_base/lib/abstract_models/common.py

```diff
@@ -14,6 +15,27 @@
 logger = logging.getLogger('ansible_base.lib.abstract_models.common')
 
 
+def get_cls_view_basename(cls):
+    # This gives the expected base Django view name of cls
+    if hasattr(cls, 'router_basename'):
+        return cls.router_basename
+    return underscore(cls.__name__)
+
+
+def get_url_for_object(obj, request=None):
+    # get_absolute_url mainly exists to support AWX
```

Comment:
This feels a little odd to me, having project-specific exceptions in DAB.

I know AWX doesn't use DAB `CommonModel` right now, but I'm wondering if maybe in the future AWX does something like having an `AWXCommonModel` that extends `CommonModel` and overrides this method, to account for the AWX specific bits, rather than doing it here. 🤔

Reply:
This isn't for linking AWX serializers to DAB models. This is for linking DAB serializers to AWX models.

So for RBAC, we have these roles, and users given a role for an object. I'll call that an assignment, a user is assigned a role, for an object. The role-assignment needs to link the object it applies for. This is an AWX object. That object has `get_absolute_url` method on it. Otherwise we _have no way_ of identifying or reasonably guessing what the view name is.

I'm all ears for if you have another solution. But DAB models are expected to link to local models over and over again if you look at what the current features are.